src/lib/orion_websocket/ws.cpp

```diff
@@ -105,9 +110,26 @@ static int wscllbck(libwebsocket_context *ctx,
 
     case LWS_CALLBACK_RECEIVE:
     {
-      ConnectionInfo *ci = connection_manager_get(dat->cid, message);
-      dat->message = strdup((restService(ci, orionServices)).c_str());
-      libwebsocket_callback_on_writable(ctx, ws);
+      const size_t remaining = lws_remaining_packet_payload(ws);
+      const size_t msg_size = strlen(message);
+
+      if (!dat->request)
+        dat->request = (char *) malloc (msg_size + remaining);
+
+      memcpy(&dat->request[dat->index], message, msg_size);
```

Comment:
ok, I understand now. The first time (if (!dat->request)), you allocate room for the entire message, knowing the total size thanks to lws_remaining_packet_payload.
Then, in consequent calls you just add to dat->request until all the message is received which is when dat->request is set to NULL.

NTC